services/core/IEEE_2030_5/ieee_2030_5/agent.py

```diff
@@ -0,0 +1,253 @@
+        self._server_usage_points: m.UsagePointList
+        self._client = IEEE_2030_5_Client(cafile=self._cacertfile,
+                                         server_hostname=self._server_hostname,
+                                         keyfile=self._keyfile,
+                                         certfile=self._certfile,
+                                         server_ssl_port=self._server_ssl_port,
+                                         pin=self._pin)
+        
+        # Hook up events so we can respond to them appropriately        
+        self._client.der_control_event_started(self._control_event_started)
+        self._client.der_control_event_ended(self._control_event_ended)
+        
+        self._client.start()
+        _log.info(self._client.enddevice)
+        assert self._client.enddevice
+        self._point_to_reading_set: Dict[str, str] = {}
+        self._mirror_usage_points: Dict[str, m.MirrorUsagePoint] = {}
+        self._mup_readings: Dict[str, m.MirrorMeterReading] = {}
+        self._mup_pollRate: int = 60
+        self._times_published: Dict[str, int] = {}
+                
+        
+        # Set a default configuration to ensure that self.configure is called immediately to setup
+        # the agent.
+        self.vip.config.set_default("config", self._default_config)
+        # Hook self.configure up to changes to the configuration file "config".
+        self.vip.config.subscribe(self.configure,
+                                  actions=["NEW", "UPDATE"],
+                                  pattern="config")
+    
+    @RPC.export
+    def update_der_settings(self, href: str, new_settings: m.DERSettings) -> int:
+        resp = self._client.put_der_settings(href, new_settings)
+        return resp
+    
+    @RPC.export
+    def update_der_availability(self, href: str, new_availability: m.DERAvailability) -> int:
+        resp = self._client.put_der_availability(href, new_availability)
+        return resp
+    
+    @RPC.export
+    def update_der_status(self, href: str, new_availability: m.DERAvailability) -> int:
+        resp = self._client.put_der_status(href, new_availability)
+        return resp
+        
+    @RPC.export
+    def get_der_references(self) -> List[str]:
+        return self._client.get_der_hrefs()
+    
+    def _control_event_started(self, sender, **kwargs):
+        _log.debug(f"Control event started {kwargs}")
+    
+    def _control_event_ended(self, sender, **kwargs):
+        _log.debug(f"Control event ended {kwargs}")
+    
+    def dcap_updated(self, sender):
+        _log.debug(f"Dcap was updated by {sender}")
+
+    def configure(self, config_name, action, contents):
+        """
+        Called after the Agent has connected to the message bus. If a configuration exists at startup
+        this will be called before onstart.
+
+        Is called every time the configuration in the store changes.
+        """
+        config = self._default_config.copy()
```

Comment:
Configure method is not updating self._mirror_usage_point_list and default_config points to that.

Reply:
On line 176 they are
self._mup_readings.clear()
self._mirror_usage_points.clear()

ed = self._client.end_device
self._mirror_usage_points.update(new_usage_points)
server_usage_points = self._client.mirror_usage_point_list()
self._mup_pollRate = server_usage_points.pollRate if server_usage_points.pollRate else self._mup_pollRate